### DISCUSSION TITLE
Is it possble to limit users to their own items?

### DISCUSSION THREAD

**Discussion by armanm** [directus/directus#6926] [Archive]
Let me start by saying Directus looks amazing and the devs deserve a huge thanks 🙇‍♂️ 

I'm evaluating whether it makes sense to user Directus as a portal for our clients to provide us with collections of data. When reading the docs, it's not clear to me whether within the CMS, all users with the same role can access each other's items or their own items only?

Our objective is to let our users provdide their own catalog of products (lets call it an items collection) for us to publish on an e-commerce website. We also want each user to have their own profile page (so that can be configured as singleton collection). If I define a role called `supplier` and assign it to every new user, is there a way to make sure each user can only see/edit his own items? same thing for the singleton profile collection?

Thanks

**Comment by rijkvanzanten** [comment]
Hi @armanm! Thanks for the kind words 🙂 

If you add a `user_created` field to your collection that holds the ID of the current user, you can then set the (read) permission of the role to `{ "user_created": { "_eq": "$CURRENT_USER" }}`

**Reply by armanm** [reply]
🙇

**Reply by armanm** [reply]
@rijkvanzanten there are still a few things I can not understand. Maybe the way I'm thinking about Directus isn't the right way 😕 

* How do you remove unnecessary modules for non admin roles? i.e. how can I stop non admins from seeing the `User Directory` or the `Documentation modules?

* It seems once you mark a collection as singleton there can only be a single record in that collection globally. Is it possible to create a singleton collection such that every user can create only 1 record of that collection in their own profile?

**Reply by rijkvanzanten** [reply]
> How do you remove unnecessary modules for non admin roles? i.e. how can I stop non admins from seeing the User Directory or the `Documentation modules?

You can override the whole nav bar under the role settings

> It seems once you mark a collection as singleton there can only be a single record in that collection globally. Is it possible to create a singleton collection such that every user can create only 1 record of that collection in their own profile?

This is correct, a singleton collection acts as a single item across the instance. You can't create a singleton where each user has its own unique copy of the row at this time
